pkg/credstash/backend.go

```diff
@@ -0,0 +1,172 @@
+// Package credstash implements backend for Credstash (that uses AWS KMS and DynamoDB)
+// Heavily inspired in github.com/ouzi-dev/credstash-operator using https://github.com/versent/unicreds
+package credstash
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/containersolutions/externalsecret-operator/pkg/backend"
+	"github.com/containersolutions/externalsecret-operator/pkg/utils"
+	unicreds "github.com/versent/unicreds"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	defaultRegion          = "eu-west-2"
+	credstashVersionLength = 19
+)
+
+var (
+	log                     = ctrl.Log.WithName("credstash")
+	table                   = ""
+	configEncryptionContext = make(map[string]string)
+)
+
+// SecretManagerClient will be our unicreds client
+type SecretManagerClient interface {
+	SetKMSConfig(config *aws.Config)
+	SetDynamoDBConfig(config *aws.Config)
+	GetHighestVersionSecret(tableName *string, name string, encContext *unicreds.EncryptionContextValue) (*unicreds.DecryptedCredential, error)
+	GetSecret(tableName *string, name string, version string, encContext *unicreds.EncryptionContextValue) (*unicreds.DecryptedCredential, error)
+}
+
+// SecretManagerClientStruct defining this struct to write methods for it
+type SecretManagerClientStruct struct {
```

Comment:
IMHO, using the `type` of an object on its own name is usually considered a code smell :< 

What do you think about renaming it to `SMC`?

Reply:
I think SMC is good enough :smiley:. unicreds does not provide a struct containing the client methods that I need from them... That is why I had to create my own to throw inside our interface on our side. Let me know if there is any better way to do this.

Reply:
Changing the interface to SecretManagerClientProvider and the struct to SecretManagerClient. Looked over some other soced around and this seems to be something that people do :)

Reply:
Perfect :D